include/boost/numeric/ublas/tensor/detail/extents_functions.hpp

```diff
@@ -220,7 +219,7 @@ template <class ExtentsType>
 inline 
 constexpr bool is_scalar(ExtentsType const &e) {
 
-  static_assert(detail::is_extents<ExtentsType>::value, "boost::numeric::ublas::is_scalar() : invalid type, type should be an extents");
+  static_assert(is_extents_v<ExtentsType>, "boost::numeric::ublas::is_scalar() : invalid type, type should be an extents");
 
   using size_type = typename ExtentsType::size_type;
   using value_type = typename ExtentsType::value_type;
```

Comment:
I recommend to use:
```cpp
return !e.empty() && std::all_of(e.begin(), e.end(), [](auto a) { return a==1u; });
```

Reply:
this was to make code more readable.

Reply:
and I can not use `1u` because we do not know the type of extents.

Reply:
but you know that 
```is_unsigned_v<ExtentsType::size_type)>```
and
```is_unsigned_v<ExtentsType::value_type)>```
So comparing with values with `1u` should always work even if number of bits differ.

Reply:
I was also considering if a user created his own extents he might make `signed` extents.

Reply:
Valid point. However, I see 
```cpp
is_unsigned_v<typename extents_traits<extents_t>::size_type> == true
```
a necessity, therefore a `static_assert` and if 
```cpp
is_unsigned_v<typename extents_traits<extents_t>::size_type> == false
```
the user gets a warning which is totally fine. Signed extents or negative extents do not make much sense. However, the user might have chosen signedness because of carelessness or special behavior for negative numbers. 

So I am okay if the user gets a warning.

Reply:
I will change it.